---
title: "HW 1 for Git upload"
author: "Katie"
date: "9/25/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse) 
```

```{r create_dataframe_one}
set.seed(1) 

dataframe_one = tibble(
  random_sample = rnorm(8), 
  logical_vector = random_sample > 0, 
  character_vector = c("I", "sometimes", "like", "to", "eat", "cereal", "for", "dinner"),
  factor_vector = factor(c("low", "medium", "high", "low", "medium", "high", "low", "medium")) 
)

mean_of_random_sample = mean(pull(dataframe_one, random_sample))
mean_of_logical_vector = mean(pull(dataframe_one, logical_vector))
mean_of_character_vector = mean(pull(dataframe_one, character_vector)) ## Error occurred when running this line of code
mean_of_factor_vector = mean(pull(dataframe_one, factor_vector)) ## Error occurred when running this line of code
```

When trying to take the mean of each variable in our dataframe, it works for the random_sample and the logical_vector. It does not work for the character_vector or the factor_vector. 

```{r eval = FALSE, convert_to_numeric}
as.numeric(pull(dataframe_one, logical_vector))
as.numeric(pull(dataframe_one, character_vector))
as.numeric(pull(dataframe_one, factor_vector))
```

When we apply the as.numeric function to the logical, character, and factor variables, we are able to turn the logical and factor variables into numbers. However, we aren't able to turn the character variables into numbers. 

